File: Common/Models/SensorWithValuesModel.cs
namespace Common.Models
{
    public class SensorWithValuesModel
    {
        public int Id { get; set; }

        public int PositionIndex { get; set; }

        public int DecimalPlaces { get; set; }

        public string MeasuringUnitName { get; set; } = string.Empty;

        public int ModuleId { get; set; }
        public ICollection<SensorValueModel> sensorValues { get; set; } = new List<SensorValueModel>();
    }
}

File: SPA/ClientApp/src/app/models/SensorWithValuesModel.ts
import { SensorValueModel } from "./SensorValueModel";

export class SensorWithValuesModel {
    id!: number;
    positionIndex: number = 0;
    decimalPlaces: number = 3;
    measuringUnitName: string = 'unit not set';
    moduleId!: number;
    sensorValues: SensorValueModel[] = [];
}